resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -419,6 +422,30 @@ private[yarn] class YarnAllocator(
     }
   }
 
+  private def decommissionNodes(allocateResponse: AllocateResponse): Unit = {
+    if (sparkConf.get(YARN_EXECUTOR_DECOMMISSION_ENABLED)) {
+      try {
+        allocateResponse.getUpdatedNodes.asScala.filter(shouldDecommissionNode).
+          foreach(node => driverRef.send(DecommissionExecutorsOnHost(getHostAddress(node))))
+      } catch {
+        case e: Exception => logError("Node decommissioning failed", e)
+      }
+    }
+  }
+
+  private def shouldDecommissionNode(nodeReport: NodeReport): Boolean = {
+    nodeReport.getNodeState match {
+      case NodeState.DECOMMISSIONING =>
+        true
+      case _ =>
+        false
+    }
+  }
+
+  private def getHostAddress(nodeReport: NodeReport): String = {
+    new URI(s"http://${nodeReport.getHttpAddress}").getHost
```

Comment:
What about `getNodeId.getHost`?

https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/yarn/api/records/NodeReport.html#getNodeId--
https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/yarn/api/records/NodeId.html#getHost--

Reply:
and you can inline this method too